### ISSUE TITLE
new mac signing issue?

### ISSUE THREAD

**Issue by madilynpaul** [mphowardlab/gsd-vmd#15]
On my my new arm64 mac, I get this error when loading the GSD plugin:
```
Warning) Unable to open dynamic library '/Applications/VMD 1.9.4a57-arm64-Rev12.app/Contents/vmd/plugins/MACOSXARM64/molfile/gsdplugin.so'.
Warning) dlopen(/Applications/VMD 1.9.4a57-arm64-Rev12.app/Contents/vmd/plugins/MACOSXARM64/molfile/gsdplugin.so, 0x0002): tried: '/Applications/VMD 1.9.4a57-arm64-Rev12.app/Contents/vmd/plugins/MACOSXARM64/molfile/gsdplugin.so' (code signature in <54E6F370-DEEA-3B53-AF1B-D4414714853C> '/Applications/VMD 1.9.4a57-arm64-Rev12.app/Contents/vmd/plugins/MACOSXARM64/molfile/gsdplugin.so' not valid for use in process: mapping process and mapped file (non-platform) have different Team IDs), '/System/Volumes/Preboot/Cryptexes/OS/Applications/VMD 1.9.4a57-arm64-Rev12.app/Contents/vmd/plugins/MACOSXARM64/molfile/gsdplugin.so' (no such file), '/Applications/VMD 1.9.4a57-arm64-Rev12.app/Contents/vmd/plugins/MACOSXARM64/molfile/gsdplugin.so' (code signature in <54E6F370-DEEA-3B53-AF1B-D4414714853C> '/Applications/VMD 1.9.4a57-arm64-Rev12.app/Contents/vmd/plugins/MACOSXARM64/molfile/gsdplugin.so' not valid for use in process: mapping process and mapped file (non-platform) have different Team IDs)
Warning) Unable to dlopen '/Applications/VMD 1.9.4a57-arm64-Rev12.app/Contents/vmd/plugins/MACOSXARM64/molfile/gsdplugin.so':
  Unable to dlopen plugin file /Applications/VMD 1.9.4a57-arm64-Rev12.app/Contents/vmd/plugins/MACOSXARM64/molfile/gsdplugin.so
```
Is this something I need to fix on my end with signing the plugin somehow (this is new to me), or is this something to fix on the developer side? Thx!

**Comment by mphoward** [comment]
I haven't tried to compile this plugin on a Mac recently, but I would guess this is something that has to be done at compile-time, at the very end. It might be possible to do this with the XCode `codesign` utility, but I don't know what all goes into doing that. I would welcome a suggestion or fix if someone knows the solution.

**Comment by erjank** [comment]
(waiting to hear back from VMD email list)

**Comment by mphoward** [comment]
Thanks! Please let us know what you learn from them. If this requires a change in the VMD distribution, another option is to discuss adding this plugin internally. I never tried to do that because I wrote this a few years ago when GSD was new and  it was always easy enough to build and install downstream, but it would be convenient to have GSD work out of the box in VMD now that it's been around for a while.

**Comment by chrisjonesBSU** [comment]
> (waiting to hear back from VMD email list)

Did you ever hear back about this, or find a work around? I'm still having the same issue with VMD 2.0 and an M4 MacBook.

**Comment by mphoward** [comment]
I think most users will be happier if `gsd-vmd` comes with VMD out of the box! So in my opinion, there is no need to go through the pain of figuring out how to resolve the external signing issue. Linux users can still compile externally if they are attached to a version of VMD without GSD included.

**Comment by chrisjonesBSU** [comment]
I agree, including gsd-vmd with VMD would be ideal if it's possible. It seems to work perfectly fine with VMD 2.0 after testing it on my Linux laptop. I've also looped in @daico007 about possibly including it. He's a previous MosDeF developer who I believe is working on VMD now.

**Comment by chrisjonesBSU** [comment]
2.0b1 works for me as well (M4 MBP, macOS Tahoe 26.3.1). I did have to specify some paths in the build command, which I usually don't on Linux (or older macs)

```
cmake -S . -B build -DCMAKE_BUILD_TYPE=Release \
  -DVMD_PLUGIN_INCLUDE_PATH=/Applications/VMD2b1.app/Contents/vmd2b1/plugins/include \
  -DVMD_PLUGIN_MOLFILE_PATH=/Applications/VMD2b1.app/Contents/vmd2b1/plugins/MACOSXARM64/molfile
```

**Comment by mphoward** [comment]
It looks like we will need to change the hinting structure if that is what worked for you:
https://github.com/mphowardlab/gsd-vmd/blob/d2f7ceaaeb078e384d4720f10961020ff8931def/CMake/FindVMD.cmake#L25

Just for my knowledge: did setting `VMDDIR` as a hint not work?

**Comment by chrisjonesBSU** [comment]
TBH, I glossed over that part of the instructions and didn't try that!
